bootstrap/019_query_history_updates.sql

```diff
@@ -67,20 +63,32 @@ BEGIN
             let where_clause_complete varchar := (select 'END_TIME <> to_timestamp_ltz(\'' || :newest_completed || '\')');
             let new_closed number;
             call internal.generate_insert_statement('INTERNAL_REPORTING_MV', 'QUERY_HISTORY_COMPLETE_AND_DAILY', 'INTERNAL', 'RAW_QH_EVT', :where_clause_complete) into :new_closed;
-            insert into INTERNAL.TASK_QUERY_HISTORY SELECT :run_id, true, :input, OBJECT_CONSTRUCT('oldest_running', :oldest_running, 'newest_completed', :newest_completed, 'attempted_migrate', :migrate, 'migrate', :migrate1, 'migrate_INCOMPLETE', :migrate2, 'new_records', :new_records, 'new_INCOMPLETE', :new_INCOMPLETE, 'new_closed', coalesce(:new_closed, 0))::VARIANT;
+            -- Figure out the oldest row in the table
+            let range_min timestamp := (select min(end_time) from (
+                select min(end_time) as end_time from internal_reporting_mv.QUERY_HISTORY_COMPLETE_AND_DAILY
+                union all
+                select min(end_time) as end_time from internal_reporting_mv.QUERY_HISTORY_COMPLETE_AND_DAILY_INCOMPLETE
+            ));
+            output := OBJECT_CONSTRUCT('oldest_running', :oldest_running, 'newest_completed', :newest_completed, 'attempted_migrate', :migrate, 'migrate', :migrate1, 'migrate_INCOMPLETE', :migrate2,
+                'new_records', :new_records, 'new_INCOMPLETE', :new_INCOMPLETE, 'new_closed', coalesce(:new_closed, 0), 'range_min', :range_min, 'range_max', :newest_completed);
         ELSE
-            insert into INTERNAL.TASK_QUERY_HISTORY SELECT :dt, true, :input, OBJECT_CONSTRUCT('oldest_running', :oldest_running, 'newest_completed', :newest_completed, 'attempted_migrate', :migrate, 'migrate', :migrate1, 'migrate_INCOMPLETE', :migrate2, 'new_records', 0, 'new_INCOMPLETE', 0, 'new_closed', 0)::VARIANT;
+            let range_min timestamp := (select min(end_time) from (
+                select min(end_time) as end_time from internal_reporting_mv.QUERY_HISTORY_COMPLETE_AND_DAILY
+                union all
+                select min(end_time) as end_time from internal_reporting_mv.QUERY_HISTORY_COMPLETE_AND_DAILY_INCOMPLETE
+            ));
+            output := OBJECT_CONSTRUCT('oldest_running', :oldest_running, 'newest_completed', :newest_completed, 'attempted_migrate', :migrate, 'migrate', :migrate1, 'migrate_INCOMPLETE', :migrate2,
+                'new_records', 0, 'new_INCOMPLETE', 0, 'new_closed', 0, 'range_min', :range_min, 'range_max', :newest_completed);
         END IF;
         DROP TABLE RAW_QH_EVT;
         COMMIT;
 
+        CALL INTERNAL.SET_CONFIG('QUERY_HISTORY_MAINTENANCE', CURRENT_TIMESTAMP()::string);
```

Comment:
why move so not always happening? Not sure of intention.

Reply:
I think how we had it before was confusing.

Being outside of the block looked (at first glance) like we would have always called `SET_CONFIG`. However, we were always `RAISE`'ing in the `EXCEPTION` block, so we would have only called `SET_CONFIG` when we didn't catch an exception.

Putting it inside the `BEGIN` block made it more clear to me that we only call this when the block successfully executes.